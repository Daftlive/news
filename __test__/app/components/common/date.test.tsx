import { render, screen } from "@testing-library/react-native";
import DateText from "../../../../app/components/common/date";

describe("<DateText>", () => {
  describe("as default", () => {
    const date = new Date();
    const FormatDate = new Intl.DateTimeFormat("en-US", {
      dateStyle: "full",
      timeStyle: "long",
    }).format(new Date(date));
    const testRenderer = render(<DateText date={date} />);

    it("Renders successfully", () => {
      expect(testRenderer).toBeDefined();
    });

    it("renders correctly", () => {
      const activeButtonAsJson = testRenderer.toJSON();
      expect(activeButtonAsJson).toMatchSnapshot();
    });

    it("Outputs the correct text", () => {
      render(<DateText date={date} />);
      expect(screen.getByText(FormatDate)).toBeTruthy();
    });
  });
});
