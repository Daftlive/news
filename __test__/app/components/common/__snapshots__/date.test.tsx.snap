// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateText> as default renders correctly 1`] = `null`;
